app/assets/src/components/visualizations/GenomeViz.js

```diff
@@ -71,23 +71,18 @@ export default class GenomeViz {
     return [min(mins), max(maxs)];
   };
 
-  // Find the bar that is closest to hoverX, within hoverBuffer.
-  onMouseMove = () => {
-    if (this.sortedEndpoints.length == 0 || !this.endpointToDataIndex) {
-      return;
-    }
-
-    const hoverX = mouse(this.svg.node())[0];
+  // Find the bar that is closest to mouseX, within hoverBuffer.
+  getDataIndexForMouseX = mouseX => {
```

Comment:
A lot of the changes here are factoring out `getDataIndexForMouseX` into a separate function so it can be used in both `onMouseMove` and `onMouseClick`.